core/Plugin/Visualization.php

```diff
@@ -458,14 +457,35 @@ private function postDataTableLoadedFromAPI()
         }
 
         // deal w/ table metadata
+        $metadata = null;
         if ($this->dataTable instanceof DataTable) {
-            $this->metadata = $this->dataTable->getAllTableMetadata();
-
-            if (isset($this->metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
-                $this->reportLastUpdatedMessage = $this->makePrettyArchivedOnText();
+            $metadata = $this->dataTable->getAllTableMetadata();
+        } else {
+            // if the dataTable is Map
+            if ($this->dataTable instanceof DataTable\Map) {
+                // load all the data
+                $dataTable = $this->dataTable->getDataTables();
+                // find the latest key
+                foreach ($dataTable as $item) {
+                    $itemMetaData = $item->getAllTableMetadata();
+                    // if ts_created not exist skip
+                    if (empty($itemMetaData[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
+                        continue;
+                    }
+                    // initial metadata and update metadata if current is more recent
+                    if (!empty($itemMetaData[DataTable::ARCHIVED_DATE_METADATA_NAME])
+                      || strtotime($itemMetaData[DataTable::ARCHIVED_DATE_METADATA_NAME]) > strtotime($itemMetaData[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
+                        $metadata = $itemMetaData;
+                    }
```

Comment:
Does that logically still make sense? Isn't every available record of `$itemMetaData[DataTable::ARCHIVED_DATE_METADATA_NAME]` overwriting `$metadata`

```suggestion
                    // initial metadata and update metadata if current is more recent
                    if (!empty($itemMetaData[DataTable::ARCHIVED_DATE_METADATA_NAME])
                        && (
                            empty($metadata[DataTable::ARCHIVED_DATE_METADATA_NAME]) 
                            || strtotime($itemMetaData[DataTable::ARCHIVED_DATE_METADATA_NAME]) > strtotime($metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])
                        )
                    ) {
                        $metadata = $itemMetaData;
                    }
```